ci/infra/testrunner/tests/test_masters.py

```diff
@@ -0,0 +1,25 @@
+import time
+
+
+def test_remove_master(deployment, conf, platform, skuba, kubectl):
+    initial_masters = skuba.num_of_nodes("master")
+    new_node_number = initial_masters
+
+    # Make sure we have at least one extra master
+    assert platform.get_num_nodes("master") > initial_masters
+
+    # Add an extra master
```

Comment:
I'm confused here. Why are we adding a master if the purpose of the test is to remove a master? why not start with a fully deployed cluster and just remove a master? that would prevent the complexity of joining all but one master that creates the issues with the `deployment` fixture.